test-sbt/shared/src/main/scala/zio/test/sbt/BaseTestTask.scala

```diff
@@ -27,32 +36,29 @@ abstract class BaseTestTask[T](
     )
   } +!+ Scope.default
 
-  protected def run(
-    eventHandler: EventHandler,
-    spec: ZIOSpecAbstract
-  )(implicit trace: ZTraceElement): ZIO[Any, Throwable, Unit] =
+  private[zio] def run(
+    eventHandlerZ: ZTestEventHandler
+  )(implicit trace: ZTraceElement): ZIO[Any, Nothing, Unit] =
     ZIO.consoleWith { console =>
       (for {
-        _ <- ZIO.succeed("TODO pass this where needed to resolve #6481: " + eventHandler)
         summary <- spec
-                     .runSpecInfallible(FilteredSpec(spec.spec, args), args, console, runtime)
+                     .runSpecInfallible(FilteredSpec(spec.spec, args), args, console, runtime, eventHandlerZ)
         _ <- sendSummary.provideEnvironment(ZEnvironment(summary))
-        _ <- ZIO.when(summary.status == Summary.Failure)(
```

Comment:
Now that we're reporting the test failures to SBT, we don't have to explicitly fail here in order for the SBT Task to fail. This has been moved to a more narrow path in `ZIOSpecAbstract`, to preserve the failing behavior when the Spec is executed as a standalone ZIOApp